### ISSUE TITLE
Flaky sdsv3 test TestStreamSecrets/Default_All_bundles:_RootCA

### ISSUE THREAD

**Issue by rturner3** [spiffe/spire#3275]
```
--- FAIL: TestStreamSecrets (0.27s)
    --- FAIL: TestStreamSecrets/Default_All_bundles:_RootCA (0.00s)
        handler_test.go:632: 
            	Error Trace:	/home/runner/work/spire/spire/pkg/agent/endpoints/sdsv3/assertions.go:50
            	            				/home/runner/work/spire/spire/pkg/agent/endpoints/sdsv3/handler_test.go:632
            	Error:      	Status code="Internal" msg="malformed header: missing HTTP status; malformed header: missing HTTP content-type" does not match code="Internal" msg="unable to use \"SPIFFE validator\" on Envoy below 1.17"
            	Test:       	TestStreamSecrets/Default_All_bundles:_RootCA
FAIL
```
Encountered in a CI run unrelated to Envoy: https://github.com/spiffe/spire/runs/7527694088?check_suite_focus=true

**Comment by rturner3** [comment]
This is related to #3229. It appears to be an issue that can happen with any of the unit tests in `pkg/agent/endpoints/sdsv3` and is caused by a bug in grpc-go: https://github.com/grpc/grpc-go/issues/5511

**Comment by MarcosDY** [comment]
[grpc/grpc-go/issues/5511](https://github.com/grpc/grpc-go/issues/5511) was released on [grpc/grpc-go/pull/5513](https://github.com/grpc/grpc-go/pull/5513) and released on grpc v1.49.0.
